src/thunks/searchThunk.ts

```diff
@@ -0,0 +1,73 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import { treeherderBaseURL } from '../common/constants';
+import type { Repository, Revision } from '../types/state';
+
+export const fetchRecentRevisions = createAsyncThunk<
+  Revision[],
+  Repository['name'],
+  { rejectValue: string }
+>('search/fetchRecentRevisions', async (repository, { rejectWithValue }) => {
+  let response;
+  try {
+    response = await fetch(
+      `${treeherderBaseURL}/api/project/${repository}/push/`,
+    );
+  } catch (err) {
+    return rejectWithValue((err as Error).message);
```

Comment:
`(err as Error)` added to avoid type error

Reply:
(for another PR) I wonder if this try/catch is useful here -- my expectation is that this would be "rejected with value" with the error automatically if the fetch call throws. Although I see you force the rejectValue to be a string, but maybe this should be an Error? Up to you actually.

(another PR too) Also note the `response.json()` call below may throw as well (if the response's body isn't valid json), so it might make sense to put it inside the try/catch block.